huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go

```diff
@@ -0,0 +1,347 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/networking/v1/security/securitygroups"
+	"github.com/chnsz/golangsdk/openstack/networking/v1/subnets"
+	"github.com/chnsz/golangsdk/openstack/networking/v1/vpcs"
+	"github.com/chnsz/golangsdk/openstack/sms/v3/templates"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var AutoCreate string = "autoCreate"
+
+// ResourceServerTemplate is the impl of huaweicloud_sms_server_template
+func ResourceServerTemplate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServerTemplateCreate,
+		ReadContext:   resourceServerTemplateRead,
+		UpdateContext: resourceServerTemplateUpdate,
+		DeleteContext: resourceServerTemplateDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"region": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"project_id"},
+				Computed:     true,
+			},
+			"project_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"region"},
+				Computed:     true,
+			},
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"subnet_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"security_group_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"volume_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "SAS",
+				ValidateFunc: validation.StringInSlice([]string{
+					"SAS", "SSD", "GPSSD", "ESSD",
+				}, true),
+			},
+			"flavor": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"target_server_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"vpc_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildVpcOpts(client *golangsdk.ServiceClient, vpcID string) (*templates.VpcRequest, error) {
+	if vpcID != "" && vpcID != AutoCreate {
```

Comment:
pls describe `autoCreate` in the document

Reply:
done